[Mappings editor] Add scaling_factor as a required parameter

alisonelizabeth opened the pull request:
This PR makes `scaling_factor` a required parameter for the `scaled_float` data type.

Fixes https://github.com/elastic/kibana/issues/52389

<img width="1202" alt="Screen Shot 2019-12-11 at 4 07 19 PM" src="https://user-images.githubusercontent.com/5226211/70660514-6400bf80-1c30-11ea-8ea5-f492fbf8ff9d.png">

<img width="721" alt="Screen Shot 2019-12-11 at 4 07 34 PM" src="https://user-images.githubusercontent.com/5226211/70660516-6400bf80-1c30-11ea-9ac6-082338c87dc2.png">

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/)
* Commit: c0a84b1fb4e8ab29973713642a77444ac475f7ad

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/","number":"14959","commit":"c0a84b1fb4e8ab29973713642a77444ac475f7ad"}]}
PIPELINE-->

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/numeric_type.tsx:
```diff
@@ -44,6 +44,45 @@ interface Props {
+                description={i18n.translate(
+                  'xpack.idxMgmt.mappingsEditor.scalingFactorFieldDescription',
+                  {
+                    defaultMessage: 'The scaling factor to use when encoding values.',
```
Seems a little strange that this text is different than the text you see when you first create the field. Can we make them the same?

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/numeric_type.tsx:
```diff
@@ -44,6 +44,45 @@ interface Props {
+                <UseField
+                  path="scaling_factor"
+                  config={getFieldConfig('scaling_factor')}
+                  component={RangeField}
```
I think this should just be a number input. The range has a min and max, but the docs don't indicate that there is a maximum value for this field. One of the examples has a scaling factor of 100.

alisonelizabeth replied:
Good point. I agree a number input makes more sense here.

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/parameters_definition.ts:
```diff
@@ -304,18 +304,42 @@ export const PARAMETERS_DEFINITION = {
       ],
+      helpText: i18n.translate('xpack.idxMgmt.mappingsEditor.parameters.scalingFactorHelpText', {
+        defaultMessage:
+          'Values will be multiplied by this factor at index time and rounded to the closest long value.',
```
I think we should remove this help text and move it into the description instead. It seems like the help text is better used for describing validation rules. If you agree I'll update https://github.com/elastic/kibana/pull/52363 accordingly since I did something similar there.

Here's what I think would make sense for description text:

> Values will be multiplied by this factor at index time and rounded to the closest long value. High factor values improve accuracy, but also increase space requirements.

alisonelizabeth replied:
>  It seems like the help text is better used for describing validation rules. 

This makes sense. I think I recall design suggesting the same.

alisonelizabeth commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/parameters_definition.ts:
```diff
@@ -313,19 +313,52 @@ export const PARAMETERS_DEFINITION = {
     } as FieldConfig,
   },
   scaling_factor: {
+    title: i18n.translate('xpack.idxMgmt.mappingsEditor.parameters.scalingFactorFieldTitle', {
```
how do you feel about the `title` and `description` being defined here?

alisonelizabeth commented:
@cjcenizal thanks for the review! I addressed your feedback.

Updated screenshots:
<img width="1219" alt="Screen Shot 2019-12-12 at 9 29 48 AM" src="https://user-images.githubusercontent.com/5226211/70723158-70375c00-1cc6-11ea-89f5-7f38bcc1903d.png">
<img width="707" alt="Screen Shot 2019-12-12 at 9 30 02 AM" src="https://user-images.githubusercontent.com/5226211/70723160-70375c00-1cc6-11ea-9062-472d7e80731f.png">

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15129/)
* Commit: 2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1

### History
* :broken_heart: [Build #14959](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/) failed c0a84b1fb4e8ab29973713642a77444ac475f7ad

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/","number":"14959","commit":"c0a84b1fb4e8ab29973713642a77444ac475f7ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15129/","number":"15129","commit":"2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1"}]}
PIPELINE-->

alisonelizabeth commented:
retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15187/)
* Commit: 2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1

### History
* :broken_heart: [Build #15129](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15129/) failed 2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1
* :broken_heart: [Build #14959](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/) failed c0a84b1fb4e8ab29973713642a77444ac475f7ad

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14959/","number":"14959","commit":"c0a84b1fb4e8ab29973713642a77444ac475f7ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15129/","number":"15129","commit":"2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15187/","number":"15187","commit":"2123a2a1c88f66bad5083ea81dc24e26d1bcb6a1"}]}
PIPELINE-->
